### PR TITLE
Index CCMs - Closes #1120

### LINKED ISSUE #1120: Index CCMs
### Description

To allow users to filter the responses, it is necessary to index the necessary information concerning the CCMs.

### Acceptance Criteria

- The ‘ccm’ index is created
- All the CCMs are indexed on new block
- CCM status is updated
- Cross-chain transaction status is updated

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1120 

### How was it solved?

- [x] Already addressed with the issue https://github.com/LiskHQ/lisk-service/issues/1111
  - [x] The ‘ccm’ index is created
  - [x] All the CCMs are indexed on new block
  - [x] CCM status is updated
  - [x] Cross-chain transaction status is updated
- [x] Add generic implementation for delete block
- [x] Add generic implementation for revert transaction
- [x] Fix issue with fee estimator
- [x] Fix issue with coordinator
- [x] Fix issue with indexer (Indexing blocks)
- [x] Fix issue with pending transactions cache

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-coordinator/shared/scheduler.js [hunk 1]
```diff
@@ -94,7 +93,7 @@ const initIndexingScheduler = async () => {
 	Signals.get('newBlock').add(newBlockListener);
 
 	// Retrieve enabled modules from connector
-	setRegisteredModules(await getEnabledModules());
+	registeredLiskModules = await getRegisteredModules();
```
Comment (ManuGowda) [inline services/blockchain-coordinator/shared/scheduler.js:96]: the `setRegisteredModules` logic is removed? we dont need to set after getting them?

File: services/blockchain-indexer/shared/indexer/blockchainIndex.js [hunk 2]
```diff
@@ -188,63 +178,54 @@ const updateBlockIndex = async job => {
...
+		logger.warn(`Error occured while deleting block with ID: ${block.blockHeader.id}. Will retry later.`);
 		throw error;
 	}
 };
 
 // Initialize queues
 const indexBlocksQueue = Queue(config.endpoints.cache, 'indexBlocksQueue', indexBlock, 30);
 const updateBlockIndexQueue = Queue(config.endpoints.cache, 'updateBlockIndexQueue', updateBlockIndex, 1);
-const deleteIndexedBlocksQueue = Queue(config.endpoints.cache, 'deleteIndexedBlocksQueue', deleteIndexedBlocks, 1);
+const deleteIndexedBlocksQueue = Queue(config.endpoints.cache, 'deleteIndexedBlocksQueue', deleteIndexedBlock, 1);
```
Comment (ManuGowda) [inline services/blockchain-indexer/shared/indexer/blockchainIndex.js:226]: i guess the logic was to delete block which you have fixed but the job names still refer plural? if it make sense update them as well to be singular

Comment (ManuGowda) [reply services/blockchain-indexer/shared/indexer/blockchainIndex.js:226]: or was it intended to achieve multiple block deletion?

File: services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/unlockToken.js [hunk 3]
```diff
@@ -21,8 +21,14 @@ const commandName = 'unlockToken';
 const applyTransaction = async (blockHeader, tx, dbTrx) => {
 };
 
+// eslint-disable-next-line no-unused-vars
+const revertTransaction = async (blockHeader, tx, dbTrx) => {
```
Comment (ManuGowda) [inline services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/unlockToken.js:25]: this logic is meant to handle the block deletion?

File: services/blockchain-indexer/shared/indexer/blockchainIndex.js (hunk 2 above)
Comment (nagdahimanshu) [reply services/blockchain-indexer/shared/indexer/blockchainIndex.js]: Good point. It is intended to support multiple block deletion. So, I updated the implementation now

File: services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/unlockToken.js (hunk 3 above)
Comment (nagdahimanshu) [reply services/blockchain-indexer/shared/indexer/transactionProcessor/dpos/unlockToken.js:25]: This logic is added to handle transactions on every delete block event. Basically, if any module specific operations needs to perform on delete block, we can add here.

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**